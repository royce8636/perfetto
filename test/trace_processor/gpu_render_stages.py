#!/usr/bin/python
# Copyright (C) 2019 The Android Open Source Project
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from os import sys, path
sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
import synth_common

trace = synth_common.create_trace()

trace.add_gpu_render_stages_stage_spec([{
    'name': 'stage 0'
}, {
    'name': 'stage 1'
}, {
    'name': 'stage 2'
}])

trace.add_gpu_render_stages_hw_queue_spec([{
    'name': 'queue 0'
}, {
    'name': 'queue 1'
}])

for i in range(1, 8):
  extra_data = {}
  render_target_handle = None
  if i % 4 != 0:
    extra_data['keyOnlyTest'] = None
    if i % 2 != 0:
      extra_data['stencilBPP'] = '1'
    extra_data['height'] = str(pow(i, 2))

  trace.add_gpu_render_stages(
      ts=i * 10,
      event_id=i,
      duration=5,
      hw_queue_id=i % 2,
      stage_id=i % 3,
      context=42,
      extra_data=extra_data)

# Test stage naming with render target handle.
trace.add_gpu_render_stages(
    ts=80, event_id=8, duration=5, hw_queue_id=0, stage_id=2, context=42)

trace.add_gpu_render_stages(
    ts=90,
    event_id=9,
    duration=5,
    hw_queue_id=0,
    stage_id=0,
    context=42,
    render_target_handle=0x10)

trace.add_vk_debug_marker(
    ts=91, pid=100, vk_device=1, obj=0x10, obj_name="frame_buffer")

trace.add_gpu_render_stages(
    ts=100,
    event_id=10,
    duration=5,
    hw_queue_id=0,
    stage_id=0,
    context=42,
    render_target_handle=0x10)

trace.add_vk_debug_marker(
    ts=101, pid=100, vk_device=1, obj=0x10, obj_name="renamed_buffer")

trace.add_gpu_render_stages(
    ts=110,
    event_id=11,
    duration=5,
    hw_queue_id=0,
    stage_id=0,
    context=42,
    render_target_handle=0x10)

print(trace.trace.SerializeToString())
