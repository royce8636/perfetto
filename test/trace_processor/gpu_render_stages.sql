SELECT track.name AS track_name, ts,dur, gpu_slice.name AS slice_name,
    depth, gpu_slice.arg_set_id, flat_key, string_value, gpu_slice.context_id,
    render_target, submission_id, hw_queue_id
FROM gpu_track
LEFT JOIN track USING (id)
INNER JOIN gpu_slice on gpu_track.id=gpu_slice.track_id
LEFT JOIN args ON gpu_slice.arg_set_id = args.arg_set_id
ORDER BY ts;
