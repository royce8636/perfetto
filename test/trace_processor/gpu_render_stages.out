"track_name","track_desc","ts","dur","slice_name","depth","flat_key","string_value","context_id","render_target","submission_id","hw_queue_id"
"queue 1","[NULL]",10,5,"stage 1",0,"description","stage desc 1",42,0,0,1
"queue 1","[NULL]",10,5,"stage 1",0,"keyOnlyTest","[NULL]",42,0,0,1
"queue 1","[NULL]",10,5,"stage 1",0,"stencilBPP","1",42,0,0,1
"queue 1","[NULL]",10,5,"stage 1",0,"height","1",42,0,0,1
"queue 0","queue desc 0",20,5,"stage 2",0,"keyOnlyTest","[NULL]",42,0,0,0
"queue 0","queue desc 0",20,5,"stage 2",0,"height","4",42,0,0,0
"queue 1","[NULL]",30,5,"stage 0",0,"keyOnlyTest","[NULL]",42,0,0,1
"queue 1","[NULL]",30,5,"stage 0",0,"stencilBPP","1",42,0,0,1
"queue 1","[NULL]",30,5,"stage 0",0,"height","9",42,0,0,1
"queue 0","queue desc 0",40,5,"stage 1",0,"description","stage desc 1",42,0,0,0
"queue 1","[NULL]",50,5,"stage 2",0,"keyOnlyTest","[NULL]",42,0,0,1
"queue 1","[NULL]",50,5,"stage 2",0,"stencilBPP","1",42,0,0,1
"queue 1","[NULL]",50,5,"stage 2",0,"height","25",42,0,0,1
"queue 0","queue desc 0",60,5,"stage 0",0,"keyOnlyTest","[NULL]",42,0,0,0
"queue 0","queue desc 0",60,5,"stage 0",0,"height","36",42,0,0,0
"queue 1","[NULL]",70,5,"stage 1",0,"description","stage desc 1",42,0,0,1
"queue 1","[NULL]",70,5,"stage 1",0,"keyOnlyTest","[NULL]",42,0,0,1
"queue 1","[NULL]",70,5,"stage 1",0,"stencilBPP","1",42,0,0,1
"queue 1","[NULL]",70,5,"stage 1",0,"height","49",42,0,0,1
"queue 0","queue desc 0",80,5,"stage 2",0,"[NULL]","[NULL]",42,0,0,0
"queue 0","queue desc 0",90,5,"stage 0[0x10]",0,"[NULL]","[NULL]",42,16,0,0
"queue 0","queue desc 0",100,5,"stage 0[0x10]",0,"[NULL]","[NULL]",42,16,0,0
"queue 0","queue desc 0",110,5,"stage 0[framebuffer]",0,"[NULL]","[NULL]",42,16,0,0
"queue 0","queue desc 0",120,5,"stage 0[renamed_buffer]",0,"[NULL]","[NULL]",42,16,0,0
